tests/array_list_with_map_test.c

```diff
@@ -0,0 +1,186 @@
+/**
```

Comment:
some thoughts on tests:
Some more tests on larger inputs, forcing more resizing, maybe add elements in a for loop? Then delete a bunch, and add some more?
Some invariant that should be preserved: 

random_access_set size = array_list len =  hash_table len = array_list last index (ie. no gaps).

Iterating through array list and finding each element should give you back the element.